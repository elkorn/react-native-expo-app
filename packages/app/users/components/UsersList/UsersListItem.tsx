import React from "react";
import { StyleSheet, Text, View } from "react-native";
import { User } from "../../types";
import Avatar from "./Avatar";

export default function UsersListItem({ item }: UsersListItemProps) {
  return (
    <View style={styles.listItem}>
      <Avatar src={item.avatar} />
      <View>
        <Text style={styles.name}>{item.name}</Text>
        <Text>ID: {item.id}</Text>
      </View>
    </View>
  );
}

interface UsersListItemProps {
  item: User;
}

const styles = StyleSheet.create({
  listItem: {
    paddingVertical: 12,
    paddingHorizontal: 12,
    borderBottomWidth: 0.3,
    borderColor: "#ddd",
    flex: 1,
    flexDirection: "row",
    alignItems: "center",
  },

  name: {
    fontWeight: "bold",
    fontSize: 24,
  },
});
