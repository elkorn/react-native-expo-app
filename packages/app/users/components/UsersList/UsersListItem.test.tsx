import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import UsersListItem from "./UsersListItem";

describe("components", () => {
  describe("UsersList", () => {
    describe("UsersListItem", () => {
      it("should match snapshot", () => {
        const props = {
          item: {
            id: "1",
            avatar: " https://i.pravatar.cc/150?img=1",
            name: "Morgan James",
          },
        };

        const tree = renderer.create(<UsersListItem {...props} />);

        expect(tree).toMatchSnapshot();
      });
    });
  });
});
