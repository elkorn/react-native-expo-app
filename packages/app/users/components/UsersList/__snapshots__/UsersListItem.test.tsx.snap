// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components UsersList UsersListItem should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomWidth": 0.3,
      "borderColor": "#ddd",
      "flex": 1,
      "flexDirection": "row",
      "paddingHorizontal": 12,
      "paddingVertical": 12,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 100,
        "height": 100,
        "justifyContent": "center",
        "marginRight": 12,
        "position": "relative",
        "width": 100,
      }
    }
  >
    <Image
      onLoad={[Function]}
      onLoadStart={[Function]}
      source={
        Object {
          "uri": "https://i.pravatar.cc/150?img=1",
        }
      }
      style={
        Object {
          "backgroundColor": "#eee",
          "borderRadius": 100,
          "height": 100,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "top": 0,
          "width": 100,
        }
      }
    />
    <ActivityIndicator
      animating={false}
      color="#999999"
      hidesWhenStopped={true}
      size="small"
    />
  </View>
  <View>
    <Text
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "bold",
        }
      }
    >
      Morgan James
    </Text>
    <Text>
      ID: 
      1
    </Text>
  </View>
</View>
`;
